src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,129 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Class for parsing strings into commands.
+ */
+public class Parser {
+    class EmptyInputException extends RuntimeException {
+        public EmptyInputException() {
+            super("☹ OOPS!!! The description of a todo cannot be empty.");
+        }
+    }
+
+    class InvalidInputException extends RuntimeException {

```

Comment:
Another suggestion, but it might be better if the Duke related exceptions were abstracted away into their own class